app/scripts/modules/core/src/presentation/forms/Validation.ts

```diff
@@ -0,0 +1,11 @@
+export type ValidationFunction = (value: any) => string | Function | Promise<any>;
+
+export class Validation {
+  public static compose = (...validationFns: ValidationFunction[]): ValidationFunction => {
+    return (value: any) => validationFns.reduce((error, validationFn) => error || validationFn(value), null);
+  };
+
+  public static isRequired: ValidationFunction = (val: any) => {
```

Comment:
should there be some mechanism for providing a custom message that's not `This field is required`, or are you intending to use the `message` prop on both field components to solve those use cases?

Reply:
great idea 👍

Reply:
Maybe we should just use `yup`? https://github.com/jquense/yup#string

Reply:
I added another commit which supports a custom validation message.

Reply:
I've been a bit ambivalent on `yup`. It feels like a pretty significant step into magic DSL territory and it's unclear to me how the method chaining style would feel with inline validations. On the flip side, as we're starting to build out validation constructs that will likely start getting lots of `yup`-esque assertions added over time I'm increasingly feeling more positive about it. Perhaps worth a little experimentation 🤷‍♂️